From 0e405ece942971f3413c10e9d71f775ef6dffa2b Mon Sep 17 00:00:00 2001
From: "Simental Magana, Marcos" <marcos.simental.magana@intel.com>
Date: Wed, 15 Jun 2016 10:19:02 -0500
Subject: [PATCH] stateless vncserver

---
 unix/vncserver | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/unix/vncserver b/unix/vncserver
index c4d0535..59d52b2 100755
--- a/unix/vncserver
+++ b/unix/vncserver
@@ -74,9 +74,13 @@ $defaultXStartup
        "fi\n".
        "if [ -x /etc/X11/xinit/xinitrc ]; then\n".
        "  exec /etc/X11/xinit/xinitrc\n".
+       "elif [ -x /usr/share/defaults/xinit/xinitrc ]; then\n".
+       "  exec /usr/share/defaults/xinit/xinitrc\n".
        "fi\n".
        "if [ -f /etc/X11/xinit/xinitrc ]; then\n".
        "  exec sh /etc/X11/xinit/xinitrc\n".
+       "elif [ -f /usr/share/defaults/xinit/xinitrc ]; then\n".
+       "  exec sh /usr/share/defaults/xinit/xinitrc\n".
        "fi\n".
        "[ -r \$HOME/.Xresources ] && xrdb \$HOME/.Xresources\n".
        "xsetroot -solid grey\n".
-- 
2.4.11

