From 1d2556fb31a32604cebad3915531a9ddcb53ad89 Mon Sep 17 00:00:00 2001
From: TSUKAHARA Ken <ken3@nurs.or.jp>
Date: Fri, 20 Jan 2017 00:21:39 +0900
Subject: [PATCH] Fix for shared memory leakage.

---
 vncviewer/X11PixelBuffer.cxx | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/vncviewer/X11PixelBuffer.cxx b/vncviewer/X11PixelBuffer.cxx
index 4f52cad..e23f777 100644
--- a/vncviewer/X11PixelBuffer.cxx
+++ b/vncviewer/X11PixelBuffer.cxx
@@ -187,8 +187,9 @@ int X11PixelBuffer::setupShm()
     goto free_xim;
 
   shminfo->shmaddr = xim->data = (char*)shmat(shminfo->shmid, 0, 0);
+  shmctl(shminfo->shmid, IPC_RMID, 0); // to avoid memory leakage
   if (shminfo->shmaddr == (char *)-1)
-    goto free_shm;
+    goto free_xim;
 
   shminfo->readOnly = True;
 
@@ -216,9 +217,6 @@ int X11PixelBuffer::setupShm()
 free_shmaddr:
   shmdt(shminfo->shmaddr);
 
-free_shm:
-  shmctl(shminfo->shmid, IPC_RMID, 0);
-
 free_xim:
   XDestroyImage(xim);
   xim = NULL;
