diff --git a/vncviewer/vncviewer.cxx b/vncviewer/vncviewer.cxx
index a2bc029..6923d0e 100644
--- a/vncviewer/vncviewer.cxx
+++ b/vncviewer/vncviewer.cxx
@@ -338,6 +338,11 @@ static void usage(const char *programName)
           "       %s [parameters] -listen [port] [parameters]\n",
           programName, programName);
   fprintf(stderr,"\n"
+	  "Options:\n\n"
+	  "  -display Xdisplay - Specifies the X display for the viewer window\n"
+	  "  -geometry geometry - Standard X position and sizing specification.\n");
+
+  fprintf(stderr,"\n"
           "Parameters can be turned on with -<param> or off with -<param>=0\n"
           "Parameters which take a value can be specified as "
           "-<param> <value>\n"
